pdns/pdns_recursor.cc

```diff
@@ -3798,9 +3798,16 @@ template<class T> T broadcastAccFunction(const boost::function<T*()>& func)
   }
 
   unsigned int n = 0;
-  T ret=T();
+  T ret = T();
+
   for (const auto& threadInfo : s_threadInfos) {
+    T* resp = nullptr;
     if (n++ == t_id) {
+      resp = func();
```

Comment:
I wonder if it's OK for all `broadcastAddFunction` calls. For example this change will cause the `lua-dns-script` to be loaded on the handler thread on a `reload-lua-script` command, while it's not normally loaded.

Reply:
That's a valid concern indeed. The handler thread is in some way a regular recursor thread, but in other ways not.  I have to think what's the best solution: don't be concerned the handler thread stats are not accounted for, or careful checking of all cases to see what would happen. Moving this to `Draft` for now.